packages/workflow/src/records/state-transitions.ts

```diff
@@ -255,25 +255,35 @@ export async function toUpdated(
       token
     )
 
-  const recordWithUpdatedTask = {
+  /*
+   * Figuring out which resources changes during
+   * update event is difficult so we are forwarding
+   * all the resources but we can be sure that history
+   * resources need not be included
+   */
+  const changedResources = {
     ...record,
     entry: [
-      ...record.entry.map((entry) => {
-        if (entry.resource.id !== previousTask.id) {
-          return entry
-        }
-        return {
-          ...entry,
-          resource: updatedTask
-        }
-      })
+      ...record.entry
+        .filter(
+          ({ resource: { resourceType } }) => !resourceType.endsWith('History')
+        )
```

Comment:
Is making sure history resources are not included because we don't want them to be sent back to Hearth? I'm thinking if this filtering should be more generically done at the place where we send the resources back to Hearth. Also so that this functions responsibility could stay just taking in a bundle and marking it as updated without any other logic

Reply:
The sending to hearth portion was already handled here: https://github.com/opencrvs/opencrvs-core/blob/a8640624ef885358a4170d37c39e5ae084a2bb69/packages/workflow/src/records/fhir.ts#L1108-L1116

The issue was mainly that this particular function: https://github.com/opencrvs/opencrvs-core/blob/a8640624ef885358a4170d37c39e5ae084a2bb69/packages/workflow/src/records/fhir.ts#L91-L106
sends the changed resources to hearth and then merges the response back into the changed resources. Now as inside the `sendBundleToHearth` function we are filtering out the history resources, it broke our assumption on there being a one to one relation between the changedResouces and the response received from hearth.

Reply:
The way I've understood the data models is that we deal with two bundle types:
- Our own full FHIR bundle that contains history types and that we type as a `ValidRecord`. This we use throughout the logic in workflow, search etc.
- FHIR bundle from where we filter out the non-FHIR standard *History resources. This we only convert the bundle at the point of sending it to Hearth

Is this correct? If yes, then is there a chance that the 2nd type "escapes" the toUpdated function a bit and could end up being used in logic code?

Reply:
Yes the risk is there whenever we are using the `updateFHIRBundle` function to update an existing record. We need to make sure that we are filtering out the *History resources before forwarding that to hearth